src/openms/source/ANALYSIS/ID/PeptideSearchEngineAlgorithm.cpp

```diff
@@ -476,13 +474,13 @@ void PeptideSearchEngineAlgorithm::postProcessHits_(const PeakMap& exp,
       }
 
       const MSSpectrum& exp_spectrum = spectra[scan_index];
-      FragmentIndexScorer::SpectrumMatchesTopN top_sms;
+      FragmentIndex::SpectrumMatchesTopN top_sms;
       fragment_index_.querySpectrum(exp_spectrum, top_sms); // TODO: expose top N as argument here and use report_top_hits_
 
       for (const auto& sms : top_sms.hits_)
       {
-        AASequence candidate = fragment_index_.getDB().getPeptides()[sms.peptide_idx_].sequence;
-
+        pair<size_t, size_t> candidate_snippet = fragment_index_.getPeptides()[sms.peptide_idx_].sequence;
+        AASequence candidate = AASequence::fromString(fasta_db[sms.peptide_idx_].sequence.substr(candidate_snippet.first, candidate_snippet.second));
```

Comment:
hmm is this correct if modifications are involved? now this is only the plain peptide right?

Reply:
yes, I´ll fix it.